src/quadratic/sgp.c

```diff
@@ -57,22 +57,25 @@ void cfe_sgp_free(cfe_sgp *s) {
 }
 
 void cfe_sgp_sec_key_init(cfe_sgp_sec_key *msk, cfe_sgp *sgp) {
-    cfe_vec_inits(sgp->n, &(msk->s), &(msk->t), NULL);
+    cfe_vec_inits(sgp->l, &(msk->s), &(msk->t), NULL);
 }
 
 void cfe_sgp_sec_key_free(cfe_sgp_sec_key *msk) {
     cfe_vec_frees(&(msk->s), &(msk->t), NULL);
 }
 
-void cfe_sgp_sec_key_generate(cfe_sgp_sec_key *msk, cfe_sgp *sgp) {
+void cfe_sgp_generate_sec_key(cfe_sgp_sec_key *msk, cfe_sgp *sgp) {
     cfe_uniform_sample_vec(&(msk->s), sgp->bound);
     cfe_uniform_sample_vec(&(msk->t), sgp->bound);
 }
 
-cfe_error cfe_sgp_derive_key(ECP2_BN254 *key, cfe_sgp_sec_key *msk, cfe_mat *f, cfe_sgp *sgp) {
+cfe_error cfe_sgp_derive_key(ECP2_BN254 *key, cfe_sgp *sgp, cfe_sgp_sec_key *msk, cfe_mat *f) {
     if (cfe_mat_check_bound(f, sgp->bound) == false) {
         return CFE_ERR_BOUND_CHECK_FAILED;
     };
+    if (f->rows != sgp->l || f->cols != sgp->l) {
+        return CFE_ERR_DIM_MISMATCH;
```

Comment:
Can't you use one of the other errors we have already defined?

Reply:
I changed to existent `CFE_ERR_MALFORMED_INPUT`.